configure.ac

```diff
@@ -633,7 +633,7 @@ if test x$use_hardening != xno; then
 
   if test x$TARGET_OS != xwindows; then
     AX_CHECK_COMPILE_FLAG([-fPIE],[PIE_FLAGS="-fPIE"])
-    AX_CHECK_LINK_FLAG([[-pie]], [HARDENED_LDFLAGS="$HARDENED_LDFLAGS -pie"])
+    AX_CHECK_LINK_FLAG([[-Wl,-pie]], [HARDENED_LDFLAGS="$HARDENED_LDFLAGS -Wl,-pie"])
```

Comment:
Is this also correct for non-macOS? I've seen `-pie` a lot in build systems but never `-Wl,-pie`.
@theuni?